QuentinTheFarmerGame/Systems/Move/MoveSystemUp/MoveSystemUp.cpp

```diff
@@ -9,50 +9,44 @@
 {
-    return [this](ECS::Registry& reg, int idxPacketEntities) {
-        updateUpPosition(reg, idxPacketEntities);
+    return [this](ECS::Registry& reg, int idxEntity) {
+        updateUpPosition(reg, idxEntity);
     };
 }
 
-void MoveSystemUp::updateUpPosition(ECS::Registry& entityManager, int idxPacketEntities)
+void MoveSystemUp::updateUpPosition(ECS::Registry& reg, int idxEntity)
 {
-    std::lock_guard<std::mutex> lock(entityManager._myBeautifulMutex);
+    std::lock_guard<std::mutex> lock(reg._myBeautifulMutex);
     try {
-        ECS::SparseArray<IComponent> DrawComponentArray = entityManager.get_components<IComponent>("DrawComponent");
-        if (DrawComponentArray.size() <= idxPacketEntities)
-            return;
-        DrawComponent* draw = dynamic_cast<DrawComponent*>(DrawComponentArray[idxPacketEntities].get());
-        if (!draw || !draw->draw)
-            return;
-
-        ECS::SparseArray<IComponent> PositionComponentArray = entityManager.get_components<IComponent>("Position2DComponent");
-        ECS::SparseArray<IComponent> SpeedComponentArray = entityManager.get_components<IComponent>("SpeedComponent");
-
-        if (PositionComponentArray.size() <= idxPacketEntities || SpeedComponentArray.size() <= idxPacketEntities)
-            return;
-
-        Position2DComponent* position = dynamic_cast<Position2DComponent*>(PositionComponentArray[idxPacketEntities].get());
-        SpeedComponent* speed = dynamic_cast<SpeedComponent*>(SpeedComponentArray[idxPacketEntities].get());
-
-        if (!position || !speed)
-            return;
-        if (position->y - speed->speedY < 0)
-            position->y = 0;
-        else
-            position->y -= speed->speedY;
+        ECS::SparseArray<IComponent> draws = reg.get_components<IComponent>("DrawComponent");
+        ECS::SparseArray<IComponent> positions = reg.get_components<IComponent>("Position3DComponent");
+        ECS::SparseArray<IComponent> speeds = reg.get_components<IComponent>("Speed3DComponent");
+        ECS::SparseArray<IComponent> player = reg.get_components<IComponent>("PlayerComponent");
+
+        for (ECS::entity_t entity = 0; entity < draws.size() && entity < positions.size() && entity < speeds.size() && entity < player.size(); entity++) {
+            std::shared_ptr<DrawComponent> draw = std::dynamic_pointer_cast<DrawComponent>(draws[entity]);
+            std::shared_ptr<Position3DComponent> position = std::dynamic_pointer_cast<Position3DComponent>(positions[entity]);
+            std::shared_ptr<Speed3DComponent> speed = std::dynamic_pointer_cast<Speed3DComponent>(speeds[entity]);
+            std::shared_ptr<PlayerComponent> playerComponent = std::dynamic_pointer_cast<PlayerComponent>(player[entity]);
+
+            if (!position || !speed || !draw || !playerComponent)
+                return;
+
+            position->z += speed->speedZ;
+
+            /*std::vector<std::any> values = {};

```

Comment:
May be you can remove the comments ?

Reply:
If we implement the update of the position for this new game, we'll need this line. So I'll let them commented for now since we don't know yet if we implement the updates.